spark/v3.4/spark/src/main/java/org/apache/iceberg/spark/source/BaseBatchReader.java

```diff
@@ -32,23 +32,27 @@
 import org.apache.iceberg.orc.ORC;
 import org.apache.iceberg.parquet.Parquet;
 import org.apache.iceberg.relocated.com.google.common.collect.Sets;
+import org.apache.iceberg.spark.ParquetReaderType;
 import org.apache.iceberg.spark.data.vectorized.VectorizedSparkOrcReaders;
 import org.apache.iceberg.spark.data.vectorized.VectorizedSparkParquetReaders;
 import org.apache.iceberg.types.TypeUtil;
 import org.apache.spark.sql.vectorized.ColumnarBatch;
 
 abstract class BaseBatchReader<T extends ScanTask> extends BaseReader<ColumnarBatch, T> {
   private final int batchSize;
+  private final ParquetReaderType parquetReaderType;
```

Comment:
Instead of passing this variable, let's create `BatchReadConf` and pass a reference to it here. I'd consider adding `batchReadConf()` method in `SparkReadConf` with the following fields (if you want, we can add a builder for `BatchReadConf` as well).

```
class BatchReadConf {
  int orcBatchSize() {...}
  ParquetReaderType parquetReaderType() {...}
  int parquetBatchSize() {...}
}
```

Reply:
Then we will have a reference to `BatchReadConf` in `BaseBatchReader` as opposed to having separate `batchSize`, `parquetReaderType`, and future fields we will have to add.

Reply:
Added `BatchReadConf`